src/ir/lowering.rs

```diff
@@ -592,9 +592,12 @@ impl LowerStructDefn for StructDefn {
                 self_ty,
                 fields: fields?,
                 where_clauses,
-                flags: ir::StructFlags {
-                    external: self.flags.external,
-                    fundamental: self.flags.fundamental,
+                struct_type: match (self.flags.external, self.flags.fundamental) {
+                    // A local struct is local regardless of fundamental
+                    (false, _) => ir::StructType::Local,
+                    // An external struct can be fundamental or not
+                    (true, false) => ir::StructType::External,
+                    (true, true) => ir::StructType::Fundamental,
```

Comment:
This logic directly corresponds to the way things currently work in the code. Instead of doing it with the `if local { ... } else if fundamental { ... } else { ... }` like before, we do all of that branching here.